app/Http/Controllers/APIController.php

```diff
@@ -376,10 +376,7 @@ protected function getKey()
      */
     public function transformQuerySearchText(string $search_text): string
     {
-        $formatted_search = urldecode($search_text);
-        $formatted_search = preg_replace('/[+\-><\(\)~*\"@%]+/', ' ', $formatted_search);
-        $formatted_search = preg_replace('!\s+!', ' ', $formatted_search);
-        return trim($formatted_search);
+        return transformQuerySearchText($search_text);
```

Comment:
@vichugofsl I may be misunderstanding the syntax, but this seems like it will result in a tight loop. Please explain

Reply:
@bradflood , I understand you. But, it won't be a tight loop because it is calling to the method that we have defined in the helper: `app/Helpers/Helper.php`. The case is that they (helper method and class method) have the same name.

It will be a tight loop if we would call the class method e.g.

```php
public function transformQuerySearchText(string $search_text): string
{
    return $this->transformQuerySearchText($search_text);
}
```

Reply:
thank you for the explanation